nat-lab/guidelines.md

```diff
@@ -0,0 +1,215 @@
+### Natlab Guidelines:
+
+1. **Use Parametrization:**
+   - Test across all operating systems when possible.
+   - Ensure parametrization for each node.
+
+   **Why:**
+   - Covers more areas.
+   - Better readability.
+   - Wider usage of a test.
+
+   **Example:**
+   ```
+   @pytest.mark.parametrize("alpha_node", [ConnectionTag.MAC_VM, ConnectionTag.WINDOWS_VM, ConnectionTag.DOCKER_CONE_CLIENT_1])
+   @pytest.mark.parametrize("beta_node", ConnectionTag.MAC_VM)
+   def test_params(alpha_node, beta_node):
+   ```
+
+2. **Assertions Clarity:**
+   - Assertions must be explicit and easily visible in the test.
+   - Avoid assertions in different threads, keep it within test, raise an exception instead if needed.
+
+   **Why:**
+   - Easier to track which test asserted if assert is within a test.
+   - Easier to read stacktrace.
+
+3. **Start Event Checks Promptly:**
+   - Initiate event checks before the corresponding action begins.
+
+   **Why:**
+   - Starting to wait for event after action, might result in missing the event.
+
+   **Example**:
+   - DON'T:
+   ```
+      await connect_to_vpn()
+      # event might occur event before starting to wait
+      await wait_for_event_peer(State.Connected)
+   ```
+   - DO:
+   ```
+      async with asyncio_util.run_async_context(wait_for_event_peer(State.Connected)) as event:
+            await asyncio.wait_for(
+                asyncio.gather(
+                    *[
+                        connect_to_vpn()
+                        event,
+                    ]
+                ),
+                timeout
+            )
+   ```
+
+4. **Utilize Provided Helpers:**
+   - Employ provided helper functions to establish the test environment.
```

Comment:
It's not clear which helpers are those. If you are referring to those in `helpers.py` in addition to mentioning this file, I think it would be good to add documentation to each helper.

Reply:
At the moment, there is only `helpers.py`, but in the future there might appear more stuff, so didn't want to specify. And about documentation on `helpers.py` - will do in the future

Reply:
I think that some notice about `helpers.py` file would be helpful

Reply:
Fine by me, just make sure that at least a ticket to track this is created :)

Reply:
LLT-4557

Reply:
Also added note about `helpers.py`